### PR TITLE
CMR-10147: Removing query parameters from self url when generating Child links

### PR DESCRIPTION
**Pull request by william-valencia**
The child links may contain query parameters from a get request. Need to remove these parameters when generating links in CMR-STAC

Before
Screenshot 2024-10-14 at 8 41 25 PM

After
Screenshot 2024-10-14 at 8 41 55 PM

### CODE REVIEW

File: src/routes/catalog.ts [hunk 1]
```diff
@@ -105,7 +105,7 @@ export const providerCatalogHandler = async (req: Request, res: Response) => {
   const selfLinks = generateSelfLinks(req);
   const childLinks = (collections ?? []).map(({ id, title }) => ({
     rel: "child",
-    href: `${self}/collections/${encodeURIComponent(id)}`,
+    href: `${self.replace(/\?.*$/, "")}/collections/${encodeURIComponent(id)}`,
```
Comment (dmistry1) [inline src/routes/catalog.ts:108]: Looks like we are using `.replace(/\?.*$/, "")` in a few places now, catalog.ts, rootCatalog.ts and browse.ts. We should create a util function `getBaseUrl()` that will return base url.

**Comment by macrouch** [comment]
I think this PR has the wrong title. "Self" links were not modified, only "child" links, right? I was confused by this until I referenced the ticket and saw what was going on. 

Need to make sure to update the PR title, and when Squash and Merging the PR make sure the correct message is recorded in the git history

File: src/routes/catalog.ts (hunk 1 above)
Comment (macrouch) [reply src/routes/catalog.ts:108]: I agree, this change is needed

File: src/__tests__/providerCatalog.spec.ts [hunk 2]
```diff
@@ -164,6 +164,44 @@ describe("GET /:provider", () => {
         });
       });
     });
+
+    describe(`given the provider has a collection`, () => {
+      it("has a child link for that collection without query parameters", async function () {
+        this.timeout(5000);
```
[the remark this answers is not in the document]
Comment (william-valencia) [reply src/__tests__/providerCatalog.spec.ts:170]: This was copy and pasted from the test above, but looks like I can remove it.

**Review by dmistry1 — APPROVED** [review]

**Comment by william-valencia** [comment]
> I think this PR has the wrong title. "Self" links were not modified, only "child" links, right? I was confused by this until I referenced the ticket and saw what was going on.
> 
> Need to make sure to update the PR title, and when Squash and Merging the PR make sure the correct message is recorded in the git history

Updated the title.

**Review by eudoroolivares2016 — APPROVED** [review]

**Pull request merged**